### PR TITLE
feat: add tabs RFC

### PR DESCRIPTION
**Pull request by anantkamath**
Add rfc for Tabs component

### CODE REVIEW

File: rfcs/tabs-component.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+```js
+import * as React from 'react';
+import {Tabs, TabPanel} from 'baseui/tabs';
+
+export default () => {
+  return (
+    <div>
+      <Tabs initialState={{activeKey: '1'}}>
+        <TabPanel key="1" title="Tab 1">Tab 1 Content</Tab>
```
Comment (gergelyke) [inline rfcs/tabs-component.md]: should we keep the basic usage absolute basic? so for example, leaving out the `key` prop, as that's optional?

File: rfcs/tabs-component.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+  Overrides for presentational components. See "Presentational Components Props API" below
+  * `[ComponentName]: ReactComponent | {props: {}, style: {}, component: ReactComponent}` - Optional
+* `onChange: ({activeKey: string>})` - Optional.
+  onChange handler that is called every time a new tab is selected
+* `stateReducer: (type: string, nextState: {activeKey: string}, currentState: {activeKey: string}) => stateToSet: {activeKey: string}` - Optional.
+  A state change handler.
+  * `type` - a state change type
+  * `nextState` - a new state value to be set
+  * `currentState` - current state value
+  * `stateToSet` - a return value that the state will be updated with
```
Comment (austin94) [inline rfcs/tabs-component.md]: [nit] spacing

**Comment by gergelyke** [comment]
looks great, thanks for contributing! 👍

File: rfcs/tabs-component.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+* `Tabs`
+* `TabPanel`
+* `StyledTab`
+* `StyledTabBar`
+* `StyledTabContent`
+* `ORIENTATION`
+* `STATE_CHANGE_TYPE`
+
+## `Tabs` API
```
Comment (nadiia) [inline rfcs/tabs-component.md]: We should split it into `Tabs` with `activeKey: string` prop and `StatefulTabs` with `initialState: {activeKey: string}` prop

File: rfcs/tabs-component.md [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+* `children: Array<TabPanel>` - Required.
+  An array of TabPanel items.
+* `initialState: {activeKey: string}` - Optional.
```
Comment (nadiia) [inline rfcs/tabs-component.md]: Should add what it would default to if no `activeKey` passed.

File: rfcs/tabs-component.md [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+  Initial state of the component where `activeKey` is the key of the tab to be selected by default
+* `disabled: boolean` - Optional. Default is `false`
+  If set to `true` all its tabs will be disabled
+* `orientation` - Optional. Defaults to ORIENTATION.horizontal
```
Comment (nadiia) [inline rfcs/tabs-component.md]: `orientation: $Values<typeof ORIENTATION>`

File: rfcs/tabs-component.md [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+* `overrides: {Root, Content, TabBar}` - Optional.
```
Comment (nadiia) [inline rfcs/tabs-component.md]: Above in the example we override `TabContent` should probably update example or the API description.

File: rfcs/tabs-component.md [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+
+These properties are passed to presentational (styled) components that are exported:
+
+* `$disabled: boolean` If the Tabs component (or an individual tab) is disabled
+* `$active: boolean` If a particular Tab is active
+* `$orientation: string` Orientation of the tab bar (either ORIENTATION.horizontal or ORIENTATION.vertical)
+
+## STATE_CHANGE_TYPE Constant
+
+* `activeTabChanged` - The active tab is changed (a new tab has been selected)
```
Comment (nadiia) [inline rfcs/tabs-component.md]: Maybe a shorter `change` would work here

**Comment by chasestarr** [comment]
*naming bikeshed, feel free to ignore*

the designs in figma imply that the 'tabs' component should be used for 'subnavigation'. is it valuable to name these components something explicit like `SubnaviationTabs` and `SubnavigationTab` in an attempt to suggest design best-practices? 

i can imagine a down-side to this approach would be tight coupling to current design conventions could cause inconsistencies in future iterations.

**Comment by anantkamath** [comment]
> _naming bikeshed, feel free to ignore_
> 
> the designs in figma imply that the 'tabs' component should be used for 'subnavigation'. is it valuable to name these components something explicit like `SubnaviationTabs` and `SubnavigationTab` in an attempt to suggest design best-practices?
> 
> i can imagine a down-side to this approach would be tight coupling to current design conventions could cause inconsistencies in future iterations.

My main gripe would be folks are used to looking for a simple Tabs component in  any component library. Calling it SubnavigationTabs would be be needless confusing as to what it does (is it supposed to be used only within a Navbar?), besides being perhaps (annoyingly?) too verbose.

**Comment by chasestarr** [comment]
Thanks for the contribution @anantkamath !

**Pull request merged**